### ISSUE TITLE
Intellisense slows to a crawl within minutes of application launch, as previously described

### ISSUE THREAD

**Issue by pomplesiegel** [microsoft/vscode-cpptools#6802]
I see that the issue https://github.com/microsoft/vscode-cpptools/issues/2994 has been closed, but this exact issue still occurs for me. After about 15-30 minutes the intellisense becomes totally useless (stops working, or extremely slow), and I have to restart VSCode to return normal behavior. It takes about 30 seconds to begin working again, and then it is functional for about 10-15 minutes, and we start the process over again. 

Anyone else experiencing this? I am building a very modestly sized project. 

- Mac OS 10.15.7
- VSCode 1.52.1 (extensions up to date) 
- Building Arduino projects (IDE 1.8.13) 
- Cpp properties shown below

Let me know if you need any further info. Thank you!

````
{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "/Users/me/Library/Arduino15/packages/adafruit/tools/**",
                "/Users/me/Documents/Arduino/libraries/**",
                "/Applications/Arduino.app/Contents/Java/libraries/**",
                "/Users/me/Library/Arduino15/packages/adafruit/hardware/samd/1.6.4/**"
            ],
            "forcedInclude": [
                "/Users/me/Library/Arduino15/packages/adafruit/hardware/samd/1.6.4/cores/arduino/Arduino.h"
            ],
            "defines": [
                "ARDUINO=10800",
                "USBCON"
            ],
            "macFrameworkPath": [
                "/System/Library/Frameworks",
                "/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c11",
            "cppStandard": "c++17"
        }
    ],
    "version": 4
}
````

**Comment by sean-mcmanus** [comment]
We need more info. What is the memory and CPU usage of cpptools and cpptools-srv processes when it's slow?

Can you follow the steps at https://github.com/microsoft/vscode-cpptools/wiki/Troubleshooting-Performance-Issues#macos to get profiling info for the process that is behaving poorly?

Also, we're planning to release a 1.2.0-insiders2 for Mac soon with lots of fixes, so you may want to retry with that.

**Comment by magwa101** [comment]
Version: 1.54.1
Commit: f30a9b73e8ffc278e71575118b6bf568f04587c8
Date: 2021-03-04T22:42:18.719Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 19.6.0

I attached an "instrument" trace as you instructed. How can I access the "insiders" version to test?

When "reindexing" (which happens very frequently) I get this log output and a ~50 second delay:

Shutting down IntelliSense server: /Users/magwa101/src/github.com/Udemy/Learn C++ and UE/ToonTanksProjectSetup_4.25/ToonTanks/Source/ToonTanks/Pawns/PawnTank.cpp
Checking for syntax errors: file:///Users/magwa101/src/github.com/Udemy/Learn%20C%2B%2B%20and%20UE/ToonTanksProjectSetup_4.25/ToonTanks/Source/ToonTanks/Pawns/PawnTank.h
terminating child process: 5942
terminating child process: 5943
sending compilation args for /Users/magwa101/src/github.com/Udemy/Learn C++ and UE/ToonTanksProjectSetup_4.25/ToonTanks/Source/ToonTanks/Pawns/PawnTank.cpp
 
[instrumentscli0.trace.zip](https://github.com/microsoft/vscode-cpptools/files/6117900/instrumentscli0.trace.zip)

**Comment by sean-mcmanus** [comment]
@magwa101 Your issue is unrelated to the original issue -- you should file a separate issue with your repro details. 1.2.2 is the latest version and we haven't released 1.3.0-insiders yet. What do you mean by "reindexing"? A 50 second delay sounds like it might be "expected" given the particular TU if the code is complicated enough. I didn't see anything unusual in the trace logging other than IntelliSense parsing. @Colengms has some experience with investigating Unreal Engine performance issues (see https://github.com/microsoft/vscode-cpptools/issues/5849 ).
